pytest/lib/mocknet.py

```diff
@@ -337,96 +335,108 @@ def stop_node(node):
 
 
 def create_and_upload_genesis(nodes, genesis_template_filename):
-    mocknet_genesis_filename = '/tmp/mocknet_genesis.json'
-    create_genesis_file(nodes, genesis_template_filename, mocknet_genesis_filename)
+    mocknet_genesis_file = tempfile.NamedTemporaryFile(mode='r+', delete=False, suffix='.mocknet.genesis')
```

Comment:
Your suggestions how to avoid closing the file make sense to me, and I tried them out, but I got a truncated `genesis.json`. Given that I close and reopen the files, `delete=False` is actually needed. No harm in keeping these temp files in the temp directory, it's about 6MB of files per loadtest run.

Reply:
You probably just need to call `flush` on the objected returned by `NamedTemporaryFile` before you pass the path to the file to external command (such as zip).